vendor/bat-native-ads/src/bat/ads/ad_conversion_info.cc

```diff
@@ -0,0 +1,81 @@
+/* Copyright (c) 2019 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "bat/ads/ad_conversion_info.h"
+
+#include "bat/ads/internal/json_helper.h"
+
+namespace ads {
+
+AdConversionInfo::AdConversionInfo() :
+    observation_window(0) {}
+
+AdConversionInfo::AdConversionInfo(
+    const AdConversionInfo& info)
+    : creative_set_id(info.creative_set_id),
+      type(info.type),
+      url_pattern(info.url_pattern),
+      observation_window(info.observation_window) {}
+
+AdConversionInfo::~AdConversionInfo() = default;
+
+const std::string AdConversionInfo::ToJson() const {
```

Comment:
as far as std::string is a component of a standard library we can assume it is a built-in type. You should avoid confusing the client programmer and leave off the const when returning a built-in type by value. That technique works good with user-defined types otherwise it creates unneeded confusion.

Reply:
Fixed